prow/crier/controller_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crier
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/client/clientset/versioned/fake"
+	v12 "k8s.io/test-infra/prow/client/listers/prowjobs/v1"
+	"k8s.io/test-infra/prow/kube"
+	"sync"
+	"testing"
+	"time"
+)
+
+const (
+	testTimeLimit = 1 * time.Second
+)
+
+//Fake Reporter
+//Sets: Logic; Some series of test jobs should be reported, while others shouldn't be
+//Asserts: The correct jobs have Report called on them
+type failureReporter struct {
+	passing *bool
+	failed  *bool
+}
+
+func (f failureReporter) Report(pj *v1.ProwJob) ([]*v1.ProwJob, error) {
+	if pj.Status.State == v1.FailureState {
+		*f.passing = true
```

Comment:
this is flipped? `if pj.Status.State == v1.FailureState` ->	*f.failed = true

Reply:
Maybe the state I'm having the Reporter look for is confusing, then.

The reporter is looking to see if the controller reports a Failing ProwJob. If it is, then the test is passing. If, however, any other kind of ProwJob is reported, then the test fails.

I could just as easily have it look for a pending ProwJob, or any other criteria, and I think the test would still hold